frontend/src/Order.tsx

```diff
@@ -80,184 +81,206 @@ const Order = () => {
     }
 
     const onClickCreate = async () => {
+        setIsSubmitEnabled(false)
         const orderFormString = buildOrderForm
         const form = JSON.stringify(orderFormString)
         console.log('Submitting form: ' + form)
+
         const response = await api.submitForm(form)
 
-        // Set resultRitm variable for later use in message to user
         setResultRitm(response)
 
+        setSubmitPopupOpen(true)
+
         console.log(response)
     }
 
+    const handleClose = () => {
+        setSubmitPopupOpen(false)
+    }
+
+    // This callback is called when the order is submitted and the user confirms the RITM returned
+    const onConfirmedSubmit = () => {
```

Comment:
Why not just use `handleClose` when `onConfirmedSubmit` is used?

Reply:
Because we will probably add more logic to the confirm submit in the future, and also add the case for when the order fails, at which point we will still do the "handleClose" but not the "onConfirmedSubmit"

Reply:
Ok, `onConfirmedSubmit` should be called `onConfirmSubmit`, or maybe something else since it is not really a confirmation of the submission of the form?

Reply:
That's why it has the current naming and not the standard "onConfirmSubmit"

Reply:
Will change the naming though